streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1176,7 +1187,7 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
 
         // generate producer configs from original properties and overridden maps
         final Map<String, Object> props = new HashMap<>(eosEnabled ? PRODUCER_EOS_OVERRIDES : PRODUCER_DEFAULT_OVERRIDES);
-        if ("exactly_once_beta".equals(getString(StreamsConfig.PROCESSING_GUARANTEE_CONFIG))) {
+        if (StreamThread.eosBetaEnabled(this) && !StreamThread.eosUpgradeModeEnabled(this)) {
```

Comment:
This condition seems a bit overkill -- are you just checking that eosBeta && upgrade.from is empty?

Reply:
Correct. I see your point, however, IMHO the code is easier to read and it's not on the critical code path, so I should be ok?